pkg/api/controller.go

```diff
@@ -99,6 +99,225 @@ type Controller struct {
 	otfDiffService        *tablediff.Service
 }
 
+func (c *Controller) CreatePresignMultipartUpload(w http.ResponseWriter, r *http.Request, repository string, branch string, params apigen.CreatePresignMultipartUploadParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.WriteObjectAction,
+			Resource: permissions.ObjectArn(repository, params.Path),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "create_presign_multipart_upload", r, repository, branch, "")
+
+	// check if api is supported
+	storageConfig := c.getStorageConfig()
+	if !swag.BoolValue(storageConfig.PreSignMultipartUpload) {
+		writeError(w, r, http.StatusNotImplemented, "presign multipart upload API is not supported")
+		return
+	}
+
+	repo, err := c.Catalog.GetRepository(ctx, repository)
```

Comment:
You can skip this - BranchExists will already do that for you

Reply:
Glad it is cached, we need it for `BlockAdapter.ResolveNamespace`